src/mlir/ast-utils.cc

```diff
@@ -301,9 +309,14 @@ namespace mlir::verona::ASTInterface
     return isA(ast, NodeKind::Else);
   }
 
+  bool hasElse(::ast::WeakAst ast)
+  {
+    return isA(ast, NodeKind::If) && hasA(ast, NodeKind::Else);
+  }
+
   ::ast::WeakAst getCond(::ast::WeakAst ast)
   {
-    assert(isIf(ast) && "Bad node");
+    assert(isAny(ast, {NodeKind::If, NodeKind::While}) && "Bad node");
```

Comment:
We will probably end up with a bunch of conditional and loop constructs, it might be worth factoring this check out into an isConditional or similar.

Reply:
The `isConditional` checks if the node is `cond`. This check is to make sure the node "could possibly have conditionals". For now, `if` and loops can, so I'll change it to that.